File: GeneratedEntityReaderClasses/Components/ItemAlchemyComponent.cs
namespace GeneratedNoitaClasses;
public class ItemAlchemyComponent : NoitaComponentBase,  iNoitaType<ItemAlchemyComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt material_make_always_cast { get; set; } = new();
	public NoitaInt material_remove_shuffle { get; set; } = new();
	public NoitaInt material_animate_wand { get; set; } = new();
	public NoitaInt material_animate_wand_alt { get; set; } = new();
	public NoitaInt material_increase_uses_remaining { get; set; } = new();
	public NoitaInt material_sacrifice { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		material_make_always_cast.Read(s);
		material_remove_shuffle.Read(s);
		material_animate_wand.Read(s);
		material_animate_wand_alt.Read(s);
		material_increase_uses_remaining.Read(s);
		material_sacrifice.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		material_make_always_cast.Write(s);
		material_remove_shuffle.Write(s);
		material_animate_wand.Write(s);
		material_animate_wand_alt.Write(s);
		material_increase_uses_remaining.Write(s);
		material_sacrifice.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/ItemAlchemyComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class ItemAlchemyComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    material_make_always_cast:NoitaInt = field(default_factory=NoitaInt)
    material_remove_shuffle:NoitaInt = field(default_factory=NoitaInt)
    material_animate_wand:NoitaInt = field(default_factory=NoitaInt)
    material_animate_wand_alt:NoitaInt = field(default_factory=NoitaInt)
    material_increase_uses_remaining:NoitaInt = field(default_factory=NoitaInt)
    material_sacrifice:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.material_make_always_cast.Read(s)
        self.material_remove_shuffle.Read(s)
        self.material_animate_wand.Read(s)
        self.material_animate_wand_alt.Read(s)
        self.material_increase_uses_remaining.Read(s)
        self.material_sacrifice.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.material_make_always_cast.Write(s)
        self.material_remove_shuffle.Write(s)
        self.material_animate_wand.Write(s)
        self.material_animate_wand_alt.Write(s)
        self.material_increase_uses_remaining.Write(s)
        self.material_sacrifice.Write(s)

